doc/api/next_api_changes/behavior/23525-SM.rst

```diff
@@ -0,0 +1,5 @@
+Bar plot ``label`` argument labels individual bars when length matches number of bars.
```

Comment:
Our definition of "behavior changes" makes existing user code do something different. Since passing a list of labels was not valid API before, you can leave this out.